utils/make_changelog.py

```diff
@@ -110,9 +110,29 @@ def get_commits_from_page(page):
     return commits
 
 
+# Get list of commits a specified commit is cherry-picked from. Can return an empty list.
+def parse_original_commits_from_cherry_pick_message(commit_message):
+    prefix = '(cherry picked from commits'
+    pos = commit_message.find(prefix)
+    if pos == -1:
+        prefix = '(cherry picked from commit'
+        pos = commit_message.find(prefix)
+        if pos == -1:
+            return []
+    pos += len(prefix)
+    endpos = commit_message.find(')', pos)
+    if endpos == -1:
+        return []
+    lst = [x.strip() for x in commit_message[pos:endpos].split(',')]
+    lst = [x for x in lst if x]
+    return lst
+
+
 # Use GitHub search api to check if commit from any pull request. Update pull_requests info.
-def find_pull_request_for_commit(commit_sha, pull_requests, token, max_retries, retry_timeout):
-    resp = github_api_get_json('search/issues?q={}+type:pr+repo:{}&sort=created&order=asc'.format(commit_sha, repo), token, max_retries, retry_timeout)
+def find_pull_request_for_commit(commit_info, pull_requests, token, max_retries, retry_timeout):
+    commits = [commit_info['sha']] + parse_original_commits_from_cherry_pick_message(commit_info['commit']['message'])
+    query = 'search/issues?q={}+type:pr+repo:{}&sort=created&order=asc'.format(' '.join(commits), repo)
+    resp = github_api_get_json(query, token, max_retries, retry_timeout)
```

Comment:
Without that change `find_pull_request_for_commit()` always found the release pull-request, i.e. for example, for commit https://github.com/ClickHouse/ClickHouse/commit/f0af0dcb527bce990b785593973effecd4ede93e it returned https://github.com/ClickHouse/ClickHouse/pull/7683, which doesn't contain any information for changelog. See https://github.com/ClickHouse/ClickHouse/pulls?q=is%3Apr+f0af0dcb527bce990b785593973effecd4ede93e

So I decided to add original commits for cherry-picks to that query, because https://github.com/ClickHouse/ClickHouse/pulls?utf8=%E2%9C%93&q=is%3Apr+f0af0dcb527bce990b785593973effecd4ede93e+c949eb57f5fb32a1596cbdd0f834a11d83bd1df6 returns https://github.com/ClickHouse/ClickHouse/pull/7683 and https://github.com/ClickHouse/ClickHouse/pull/8384.